src/main/java/scrolls/elder/model/DatastoreVersionStorage.java

```diff
@@ -50,7 +50,16 @@ public ReadOnlyDatastore executeRedo() {
     }
 
     public void commitDatastore(ReadOnlyDatastore datastore) {
-        this.datastoreVersions.add(datastore);
+        int size = this.datastoreVersions.size();
+
+        // If not at end of list, purge the data before adding new datastore snapshot
```

Comment:
Good catch, we might need a couple of tests for various sequences of undo and redo to catch any edge cases.

Reply:
Yep I will be implementing test cases on undo and redo commands and pushing them to this pr.